### PR TITLE
Add proofs to bytes for oois, findings and csv files

### PR DESCRIPTION
**Pull request by Donnype**
Signed-off-by: Donny Peeters <donny@bitestreams.com>

## Changes
Everywhere where we save information to Octopoes as declarations, we add a raw proof to Bytes. I also added more tests with the goal of picking up cleanup work with every new feature. The coverage in this PR goes from 75% to 78%.

## Issue ticket number and link
https://github.com/minvws/nl-kat-boefjes/issues/67

## Checklist for author(s):
- [x] This PR comes from a `feature` or `hotfix` branch, in line with our git branching strategy;
- [x] This PR is "bite-sized" and only focuses on a single issue, problem, or feature;
- [x] If a non-trivial PR: This PR is properly linked to an issue on the project board;
- [x] If a non-trivial PR: I have added screenshots or some other proof that my code does what it is supposed to do;
- [x] I am not reinventing the wheel: there is no high-quality library that already has this feature;
- [x] I have changed the example `.env` files if I added, removed, or changed any config options, and I have informed others that they need to modify their `.env` files if required;
- [x] I have performed a self-review of my own code;
- [x] I have commented my code, particularly in hard-to-understand areas;
- [x] I have made corresponding changes to the documentation, if necessary;
- [x] I have written unit, integration, and end-to-end tests for the change that I made;

```
## Checklist for functional reviewer(s):
- [ ] If a non-trivial PR: This PR is properly linked to an issue on the project board;
- [ ] I have checked out this branch, and successfully ran `make kat`;
- [ ] I have ran `make test-rf` and all end-to-end Robot Framework tests pass;
- [ ] I confirmed that the PR's advertised `feature` or `hotfix` works as intended;
- [ ] I confirmed that there are no unintended functional regressions in this branch;

### What works:
* _bullet point + screenshot (if useful) per tested functionality_

### What doesn't work:
* _bullet point + screenshot (if useful) per tested functionality_

### Bug or feature?:
* _bullet point + screenshot (if useful) if it is unclear whether something is a bug or an intended feature._
```

```
## Checklist for code reviewer(s):
- [ ] The code passes the CI tests and linters;
- [ ] The code does not bypass authentication or security mechanisms;
- [ ] The code does not introduce any dependency on a library that has not been properly vetted;
- [ ] The code does not violate Model-View-Template and our other architectural principles;
- [ ] The code contains docstrings, comments, and documentation where needed;
- [ ] The code prioritizes readability over performance where appropriate;
- [ ] The code conforms to our agreed coding standards.
```

### CODE REVIEW

**Comment by Donnype** [comment]
@underdarknl What about scan_profiles etc? Maybe we should build an Proxy to OctopoesAPIConnector that handles proofs to Bytes?

**Comment by underdarknl** [comment]
> @underdarknl What about scan_profiles etc? Maybe we should build an Proxy to OctopoesAPIConnector that handles proofs to Bytes?
Yes, but also for various other documents, like generated reports, indemnifications, gdrp docs signed by users etc.

File: tests/conftest.py [hunk 1]
```diff
@@ -80,6 +75,11 @@ def mock_views_katalogus(mocker):
     return mocker.patch("rocky.views.ooi_report.get_katalogus")
 
 
+@pytest.fixture
+def mock_bytes(mocker):
```
Comment (ammar92) [inline tests/conftest.py:79]:
```suggestion
def mock_bytes_client(mocker):
```

**Review by ammar92 — APPROVED** [review]
Looks good 👍 only one simple suggestion

File: rocky/bytes_client.py [hunk 2]
```diff
@@ -25,6 +30,71 @@ def health(self) -> ServiceHealth:
...
 
+    @staticmethod
+    def raw_from_declarations(declarations: List[Declaration]):
+        json_string = f"[{','.join([declaration.json() for declaration in declarations])}]"
+
+        return json_string.encode("utf-8")
+
+    def add_manual_proof(self, raw: bytes, manual_mime_type: str = "manual/ooi"):
+        """Per convention for a generic normalizer, we add a raw list of declarations, not a single declaration"""
+        self.login()
```
Comment (Lisser) [inline rocky/bytes_client.py:41]: Not for this PR, but let's improve the login logic later? 

I assume the retrieved token is valid for some time. So perhaps setting a TTL-cached property would do the job?

**Review by Lisser — APPROVED** [review]

**Review by underdarknl — APPROVED** [review]

**Pull request merged**